yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -135,6 +136,22 @@ private[spark] class Client(
     appContext
   }
 
+  /**
+   * Sets a node label expression that will apply to all executor containers. This is only supported
+   * in recent versions of YARN, so we need to use reflection to set this.
+   */
+  private def setNodeLabelExpressions(appContext: ApplicationSubmissionContext): Unit = {
+    sparkConf.getOption("spark.yarn.executor.nodeLabelExpression").foreach { expr =>
+      try {
+        Utils.invoke(classOf[ApplicationSubmissionContext], appContext, "setNodeLabelExpression",
+          (classOf[String], expr))
+      } catch {
+        case e: NoSuchMethodException => logInfo(s"spark.yarn.executor.nodeLabelExpression $expr " +
```

Comment:
I wonder if this should error out more catastrophically if the user is trying to use a feature that doesn't exist.